pycoshark/mongomodels.py

```diff
@@ -241,6 +242,231 @@ class CISystem(Document):
     last_updated = DateTimeField()
 
 
+class Workflow(Document):
+    """
+    `Represents a workflow in a collection.
+
+    This class inherits from :class:`~mongoengine.Document` and defines the schema for a workflow.
+
+    :property workflow_id: (:class:`~mongoengine.fields.IntField`) Unique identifier for the workflow.
+    :property name: (:class:`~mongoengine.fields.StringField`) Name of the workflow.
+    :property path: (:class:`~mongoengine.fields.StringField`) Path of the workflow (default is None).
+    :property state: (:class:`~mongoengine.fields.StringField`) Current state of the workflow.
+    :property created_at: (:class:`~mongoengine.fields.DateTimeField`) Date and time when the workflow was created.
+    :property updated_at: (:class:`~mongoengine.fields.DateTimeField`) Date and time when the workflow was last updated.
+    :property project_url: (:class:`~mongoengine.fields.StringField`) URL of the associated project (default is None).
+    :property project_id: (:class:`~mongoengine.fields.ObjectIdField`) ID of the associated project (default is None).
+    :property vcs_system_id: (:class:`~mongoengine.fields.ObjectIdField`) ID of the associated VCS system (default is None).`
+    """
+    meta = {
+        'indexes': [
+            'ci_system_id'
+        ],
+        'shard_key': ('external_id', 'ci_system_id'),
+    }
+
+    ci_system_id = ObjectIdField(required=True)
+    external_id = StringField(unique_with=['ci_system_id'])
+    name = StringField()
+    path = StringField(default=None)
+    state = StringField()
+    created_at = DateTimeField()
+    updated_at = DateTimeField()
+    project_url = StringField(default=None)
+    project_id = ObjectIdField(default=None)
+    vcs_system_id = ObjectIdField(default=None)
+
+
+class RunPullRequest(EmbeddedDocument):
```

Comment:
Is there a way to use the pull request from the prSHARK? If not, we should maybe include a way to link them IMO.

Reply:
Good idea we can do that. We will need to create a PRsystem document if we did not collect PRs for this repo befoer.
In general, there no issue doing that.

Reply:
It should only link if the pull request is already collected, if not it can print a warning that it is able to link to existing pull requests if they have been collected before.